include/dlaf/eigensolver/backtransformation.h

```diff
@@ -0,0 +1,52 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <blas.hh>
+#include "dlaf/eigensolver/backtransformation/mc.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace eigensolver {
+
+/// Eigenvalue back-transformation implementation on local memory.
+///
+/// It computes Q C, where Q = HH(1) HH(2) ... HH(m-b) (HH(j) is the House-Holder transformation defined
+/// by the j-th element of tau and the HH reflector stored in the j-th column.
+///
+/// The HH reflector are computed using the @see computeTFactor() method, starting from a matrix
+/// containing the elementary reflectors (@param mat_v) and an array of taus, associated with each
+/// reflector (@param taus). For a generic matrix @param mat_c (mxn), @param mat_v has size (mxm) and
+/// @param taus corresponds to the total number of reflectors (i.e. the number of non zero columns of
+/// @param mat_v).
+///
+/// Each tau is computed selecting a column of @param mat_v (called @p v). In case of real number, tau =
+/// 2 / (vH v), while in the complex case the real part of tau corresponds to  [1 + sqrt(1 - vH v
+/// taui^2)]/(vH v), where @p taui is complex part (random value).
```

Comment:
There is some confusion about:
- documenting the input parameters
- documenting what the output and the effects of the function
- documenting details of the methods used in the algorithm (e.g. computeTfactor)
- documenting how parameters are prepared in the test.

For the function documentation only the first two points should be covered.

In this case I suggest:
```
/// Eigenvalue back-transformation implementation on local memory.
///
/// It computes Q C, where Q = HH(1) HH(2) ... HH(m-mb) (HH(j) is the House-Holder transformation (I - v tau vH) defined
/// by the j-th element of tau and the HH reflector stored in the j-th column of the matrix V.
///
/// @param mat_c contains the (mxn) matrix C (blocksize (mbxnb)), while on exit it contains Q C.
/// @param mat_v is (mxm) matrix with blocksize (mbxmb) , which contains the Householder reflectors. The j-th HH reflector is v_j = (1, V(mb : n, j))
/// @param taus is a (blocked) vector of size m (blocksize mb). The j-th element is the scaling factor for the j-th HH tranformation.
/// @pre mat_c is not distributed,
/// @pre mat_v is not distributed.
```

Reply:
This is really clear, thank you!